File: Medii de Proiectare si Programare/HW/mpp_hw/src/main/java/mpp/com/Service/ServiceUser.java
package mpp.com.Service;

import mpp.com.Domain.User;
import mpp.com.Repository.RepoDBUser;
import mpp.com.Repository.RepositoryUser;

import java.util.Optional;

public class ServiceUser {


    RepositoryUser repoDBUser;

    public ServiceUser(RepositoryUser repoDBUser) {
        this.repoDBUser = repoDBUser;
    }

    public Optional<User> searchUsedByUsernamePassword(String username, String password) throws IllegalArgumentException {
        return repoDBUser.login(username, password);
    }

}

File: Medii de Proiectare si Programare/HW/MPP_CSharp/MPP_CSharp/Service/ServiceUser.cs
using System;
using MPP_CSharp.Domain;
using MPP_CSharp.Repository;

namespace MPP_CSharp.Service
{
    public class ServiceUser
    {
        private readonly IRepositoryUser _repoDBUser;

        public ServiceUser(IRepositoryUser repoDBUser)
        {
            _repoDBUser = repoDBUser ?? throw new ArgumentNullException(nameof(repoDBUser));
        }

        public User? SearchUserByUsernamePassword(string username, string password)
        {
            if (string.IsNullOrEmpty(username))
            {
                throw new ArgumentException("Username must not be null or empty", nameof(username));
            }

            if (string.IsNullOrEmpty(password))
            {
                throw new ArgumentException("Password must not be null or empty", nameof(password));
            }

            return _repoDBUser.FindByUsernamePassword(username, password);
        }
    }
}